docs/concepts/true-hybrid-search.mdx

```diff
@@ -0,0 +1,266 @@
+---
+title: "True Hybrid search"
+---
+
+True Hybrid Search allows you to combine multiple retrieval techniques — such as multi-vector search, hybrid (vector \+ text) search
+sort the results using a custom scoring function **within a single query**.
+
+## How TopK differs from other "hybrid" search systems
+
+Most databases that offer "hybrid" search maintain separate vector and keyword indexes. When a query is executed:
+
+1. They run the query against both indexes separately
+2. After, each index returns its top results (e.g. first 100 candidates)
+3. Lastly, use techniques like Reciprocal Rank Fusion (RRF) to merge and rerank these two sets of results
+
+<br />
+
+```mermaid
+flowchart TD
+    A[User Query] --> B[(Query Keyword Index)]
+    A --> C[(Query Vector Index)]
+
+  B --> D@{ shape: docs, label: "Top 100 Keyword search results" }
+  C --> E@{ shape: docs, label: "Top 100 Vector search results" }
+
+  D --> F{Merge and rerank top 200 results using RRF}
+  E --> F
+
+  F --> G@{ shape: stadium, label: "Final 100 ranked results" }
+```
+
+<br />
+
+This approach is fundamentally **probabilistic** - the final "top 100" results are not guaranteed to be the actual best matches because some potential matches might be missed if they don't appear in either index's top results.
+
+**TopK is different.** We scan both vector and keyword indexes simultaneously within a single query, ensuring that our "top 100" results are the **actual** top 100 - not just a probabilistic approximation:
+
+```mermaid
+flowchart TD
+    A[User Query]
+    A --> B[(Keyword Index)]
+    A --> C[(Vector Index)]
+
+    B --> D{Simultaneously scan both indexes and retrieve top 100 results}
+    C --> D
+    D --> E@{ shape: docs, label: "Combined top 100 Keyword and Vector search results" }
+    E --> F@{ shape: stadium, label: "Final 100 ranked results" }
```

Comment:
I wouldn't say "Final" here as it implies this wasn't the case all along. something like "True/Real/Actual top 100 results"